Automattic/pocket-casts-android
# Refactor code to load SubscriptionMapper using DI

## Description
This refactors code to load `SubscriptionMapper` using DI.

Fixes https://github.com/Automattic/pocket-casts-android/issues/2962

## Testing Instructions
- Make sure CI is 🟢 
- (Optional) Smoke test in the release build that trial or intro offer is loaded properly in the regular purchase flow (locked folders, bookmarks flow)

## Screenshots or Screencast 

<img width= 200 src="https://github.com/user-attachments/assets/90d999d7-fe4b-43ed-b04a-e81d79dac7ac"/>


## Checklist
 
#### I have tested any UI changes...
